src/main/java/org/mariadb/jdbc/internal/stream/PacketOutputStream.java

```diff
@@ -67,7 +59,6 @@ public void startPacket(int seqNo, boolean checkPacketLength) throws IOException
             throw new IOException("Last stream not finished");
         }
         this.seqNo = seqNo;
-        buffer.clear();
```

Comment:
How are you able to avoid resetting the buffer at these points?  This seems like an odd behavioral change on the surface.